### ISSUE TITLE
unacked message is not redelivered in setting ackTimeout

### ISSUE THREAD

**Issue by jiazhai** [apache/incubator-pulsar#2584]
#### Expected behavior

unacked message is not redelivered in setting ackTimeout

#### Actual behavior
unacked message is not redelivered in setting ackTimeout

#### Steps to reproduce
- create topic, producer and consumer, set consumer acktimeout
- call consumer.receive(), and not ack the received message.
- expected the message be redelivered after acktimeout; but actually not. 
it is actually redelivered after 2*acktimeout

#### System configuration
2.1

**Comment by jiazhai** [comment]
@codelipenghui meet this issue in their usage. 
The main reason is in UnAckedMessageTracker.
```
    public void start(PulsarClientImpl client, ConsumerBase<?> consumerBase, long ackTimeoutMillis) {
        this.stop();
        timeout = client.timer().newTimeout(new TimerTask() {
            @Override
            public void run(Timeout t) throws Exception {
                if (isAckTimeout()) {   < === first timeout, it is false, because oldOpenSet is empty.
                    log.warn("[{}] {} messages have timed-out", consumerBase, oldOpenSet.size());
                    Set<MessageId> messageIds = new HashSet<>();
                    oldOpenSet.forEach(messageIds::add);
                    oldOpenSet.clear();
                    consumerBase.redeliverUnacknowledgedMessages(messageIds);
                }
                toggle();    < === toggle after timeout
                timeout = client.timer().newTimeout(this, ackTimeoutMillis, TimeUnit.MILLISECONDS);
            }
        }, ackTimeoutMillis, TimeUnit.MILLISECONDS);
    }
```
before first timeout, all messageId was added in CurrentSet, not in OldOpenSet, so isAckTimeout() is false, and `redeliverUnacknowledgedMessages` was not called at first timeout.

The fix may move `toggle()` from behind if clause to before if clause.

**Comment by sijie** [comment]
@jiazhai can you submit a PR for that?

**Comment by sijie** [comment]
This is fixed by #2590
